clippy_lints/src/utils/mod.rs

```diff
@@ -958,7 +958,11 @@ pub fn is_refutable(cx: &LateContext<'_, '_>, pat: &Pat) -> bool {
             }
         },
         PatKind::Slice(ref head, ref middle, ref tail) => {
-            are_refutable(cx, head.iter().chain(middle).chain(tail.iter()).map(|pat| &**pat))
+            if middle.is_none() {
+                true
```

Comment:
It might still be irrefutable if `head` or `tail` are irrefutable

Reply:
Are you referring to matches on array types? As @mikerite points out, the only irrefutable slice pattern on slices is `[..]`.